core/templates/dev/head/domain/collection/ReadOnlyCollectionBackendApiService.ts

```diff
@@ -40,16 +44,27 @@ oppia.factory('ReadOnlyCollectionBackendApiService', [
 
       $http.get(collectionDataUrl).then(function(response) {
         var collection = angular.copy(response.data.collection);
+        _cacheCollectionDetails(response.data);
         if (successCallback) {
           successCallback(collection);
         }
+        $rootScope.$broadcast('collectionLoaded');
```

Comment:
I'm not sure about broadcasting this from the rootScope (And generally from the services), will there be an issue broadcasting this from the place where we are calling "loadCollection"?